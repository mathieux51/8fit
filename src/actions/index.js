export const setField = field => ({
  type: "SET_FIELD",
  field
})
