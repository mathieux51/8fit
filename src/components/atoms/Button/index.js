import React from "react"
import { TouchableOpacity } from "react-native"
import styled from "styled-components"
import PropTypes from "prop-types"
import TextMedium from "atoms/TextMedium"

const ButtonContainer = styled(TouchableOpacity)`
  background: ${({ theme, active = true, inverted = false }) =>
    active ? (inverted ? theme.white : theme.black) : theme.grey};
  border-radius: 30px;
  padding: 15px;
  min-width: 100px;
`

const StyledText = styled(TextMedium)`
  align-self: center;
  font-size: ${({ theme }) => theme.medium};
  color: ${({ theme, inverted = false }) =>
    inverted ? theme.black : theme.white};
`

const Button = props => (
  <ButtonContainer
    style={props.style}
    onPress={props.onPress}
    active={props.active}
    inverted={props.inverted}
  >
    <StyledText inverted={props.inverted}>{props.title}</StyledText>
  </ButtonContainer>
)

Button.propTypes = {
  style: PropTypes.array,
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  active: PropTypes.bool,
  inverted: PropTypes.bool
}

export default Button
