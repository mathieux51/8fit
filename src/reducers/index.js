import { combineReducers } from "redux"
import nav from "./nav"
import user from "./user"

const rootReducer = combineReducers({
  nav,
  user
})

export default rootReducer
