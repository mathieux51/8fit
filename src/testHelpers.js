import React from "react"
import { ThemeProvider } from "styled-components"
import theme from "src/styles/theme"

export function wrappeWithThemeProvider(node) {
  return <ThemeProvider theme={theme}>{node}</ThemeProvider>
}
