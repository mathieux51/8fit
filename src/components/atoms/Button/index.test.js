import React from "react"
import { shallow } from "enzyme"
import sinon from "sinon"
import { wrappeWithThemeProvider } from "src/testHelpers"
import Button from "./"

describe("<Button />", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        wrappeWithThemeProvider(<Button title="Continue" onPress={() => {}} />)
      )
    ).toMatchSnapshot()
  })

  it("renders correctly with active and inverted props", () => {
    expect(
      shallow(
        wrappeWithThemeProvider(
          <Button
            title="Continue"
            onPress={() => {}}
            active={true}
            inverted={true}
          />
        )
      )
    ).toMatchSnapshot()
  })

  it("Calls onPress when clicked", () => {
    const onPress = sinon.spy()
    const wrapper = shallow(<Button title="Continue" onPress={onPress} />)
    wrapper.first().simulate("press")
    expect(onPress.called).toBeTruthy()
  })
})
