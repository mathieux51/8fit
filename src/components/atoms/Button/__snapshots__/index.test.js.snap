// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "black": "#292C2F",
        "bold": 20,
        "boldBig": 24,
        "boldBigger": 32,
        "green": "#26D88C",
        "grey": "#959798",
        "lGrey": "#EAEAEA",
        "medium": 12,
        "regular": 16,
        "regularSmall": 12,
        "white": "#ffffff",
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Continue"
    />
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onPress": [Function],
      "title": "Continue",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onPress": [Function],
        "title": "Continue",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Button /> renders correctly with active and inverted props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "black": "#292C2F",
        "bold": 20,
        "boldBig": 24,
        "boldBigger": 32,
        "green": "#26D88C",
        "grey": "#959798",
        "lGrey": "#EAEAEA",
        "medium": 12,
        "regular": 16,
        "regularSmall": 12,
        "white": "#ffffff",
      }
    }
  >
    <Button
      active={true}
      inverted={true}
      onPress={[Function]}
      title="Continue"
    />
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "inverted": true,
      "onPress": [Function],
      "title": "Continue",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "inverted": true,
        "onPress": [Function],
        "title": "Continue",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
